salt/spm/__init__.py

```diff
@@ -245,6 +246,50 @@ def _local_install(self, args, pkg_name=None):
 
         formula_tar.close()
 
+    def _resolve_deps(self, formula_def):
+        '''
+        Return a list of packages which need to be installed, to resolve all
+        dependencies
+        '''
+        pkg_info = self.pkgdb['{0}.info'.format(self.db_prov)](formula_def['name'])
+        if not isinstance(pkg_info, dict):
+            pkg_info = {}
+
+        can_has = {}
+        cant_has = []
+        for dep in formula_def.get('dependencies', '').split(','):
+            dep = dep.strip()
+            if not dep:
+                continue
+            if self.pkgdb['{0}.info'.format(self.db_prov)](dep):
+                continue
+
+            if dep in self.avail_pkgs:
+                can_has[dep] = self.avail_pkgs[dep]
+            else:
+                cant_has.append(dep)
+
+        inspected = []
```

Comment:
since `inspected` seems that it will contain items which should be unique and the order does not matter, consider using a tuple which is way more performant than a list in checks like, `if a in <container>`, specially as it grows.